logic/transactions/pool.js

```diff
@@ -0,0 +1,922 @@
+			library.logger.debug(['Cleared duplicated transaction in pool list:', clearedList, 'transaction id:', id].join(' '));
+		}
+	}
+	return clearedList;	
+};
+
+/**
+ * Deletes transactions from pool lists and rechecks balance for ready transactions
+ * @implements {self.delete}
+ * @param {string} id transaction id
+ * @return {Array} names of cleared lists
+ */
+TransactionPool.prototype.sanitizeTransactions = function (transactions) {
+	transactions.forEach(function (transaction) {
+		self.delete(transaction.id);
+	});
+};
+
+/**
+ * Pulls transactions from unverified, perform verifications and if successful
+ * push to either verified.pending (when transaction is multisign or timestamp is in 
+ * future) or verified.ready otherwise.
+ * @implements {__private.delete}
+ * @implements {__private.processUnverifiedTransaction}
+ * @implements {self.checkBalance}
+ * @implements {__private.moveToVerified}
+ * @implements {__private.addReadyCheckBroadcast}
+ * @param {function} cb
+ * @return {setImmediateCallback} err | cb
+ */
+TransactionPool.prototype.processPool = function (cb) {
+	async.series({
+		processUnverified: function (seriesCb) {
+			if (pool.unverified.count === 0) {
+				return setImmediate(seriesCb);
+			}
+
+			async.eachSeries(pool.unverified.transactions, function (transaction, eachSeriesCb) {
+				__private.delete(transaction.id, pool.unverified);
+				__private.processUnverifiedTransaction(transaction, function (err, sender) {
+					if (err) {
+						library.logger.error('Failed to process unverified transaction: ' + transaction.id, err);
+						return setImmediate(eachSeriesCb);
+					}
+					self.checkBalance(transaction, sender, function (err, balance) {
+						if (err) {
+							library.logger.error('Failed to check balance transaction: ' + transaction.id, err);
+							return setImmediate(eachSeriesCb);
+						}
+						return __private.moveToVerified(transaction, eachSeriesCb);
+					});
+				});
+			}, function (err) {
+				return setImmediate(seriesCb, err);
+			});
+		},
+		processPending: function (seriesCb) {
+			if (pool.verified.pending.count === 0) {
+				return setImmediate(seriesCb);
+			}
+
+			// process pool.verified.pending (multisig transactions signs), and take care 
+			// about moving transactions from `verified.pending` to `verified.ready`
+			async.eachSeries(pool.verified.pending.transactions, function (transaction, eachSeriesCb) {
+				// Check multisignatures
+				if (transaction.type === transactionTypes.MULTI && 
+					Array.isArray(transaction.signatures) && 
+					transaction.signatures.length >= transaction.asset.multisignature.min
+				) {
+					__private.delete(transaction.id, pool.verified.pending);
+					__private.addReadyCheckBroadcast(transaction, eachSeriesCb);
+				} else {
+					return setImmediate(eachSeriesCb);
+				}
+			}, function (err) {
+				return setImmediate(seriesCb, err);
+			});
+		}
+	}, function (err) {
+		library.bus.message('unverifiedTransaction', pool.broadcast);
```

Comment:
Why we always fire that event regardless of error?

Reply:
because each pool cycle broadcast transactions